service/controller/app/v1/resource_set.go

```diff
@@ -159,19 +175,34 @@ func NewResourceSet(config ResourceSetConfig) (*controller.ResourceSet, error) {
 			return nil, microerror.Mask(err)
 		}
 
-		g8sClient, err := kubeConfig.NewG8sClientForApp(ctx, cr)
-		if err != nil {
-			return nil, microerror.Mask(err)
-		}
+		secretName := key.SecretName(cr)
 
-		k8sClient, err := kubeConfig.NewK8sClientForApp(ctx, cr)
-		if err != nil {
-			return nil, microerror.Mask(err)
+		var restConfig *rest.Config
+		var g8sClient versioned.Interface
+		var k8sClient kubernetes.Interface
+
+		if secretName != "" {
+			restConfig, err = kubeConfig.NewRESTConfigForApp(ctx, cr)
+			if err != nil {
+				return nil, microerror.Mask(err)
+			}
+			g8sClient, err = versioned.NewForConfig(restConfig)
+			if err != nil {
+				return nil, microerror.Mask(err)
+			}
+			k8sClient, err = kubernetes.NewForConfig(restConfig)
+			if err != nil {
+				return nil, microerror.Mask(err)
+			}
+		} else {
+			g8sClient = config.G8sClient
+			k8sClient = config.K8sClient
```

Comment:
Why are we setting the clients to something else than a tenant specific client? These are usually clients accessing the control plane. Why should we set clients differently here? I miss some background information I guess.

Reply:
In this specific case, we don't have `secret` information in `app` CR which is essential to access to tenant cluster. Therefore we regard it intended as install inside control plane.

Reply:
I see. That implementation detail can lead to pretty serious consequences. I think we should add some logging to express what is going on and why we use which client. I suppose it is a feature to work against control planes and tenant clusters in the same way. I am just wondering if it can more easily happen to miss the secret even though we intent to work against a tenant cluster.

Reply:
To me it's pretty clear. It was described in the spec. When no Secret is specified in the App CR spec then app-operator creates chart CRs using in-cluster credentials, otherwise with credentials from referenced Secret.